wallet/bip32.go

```diff
@@ -70,15 +79,49 @@ func NewMasterKeyPair(seed []byte) (*EDKeyPair, error) {
 
 func (kp *EDKeyPair) NewChildKeyPair(seed []byte, childIdx int) (*EDKeyPair, error) {
 	path := kp.Path.Extend(BIP44HardenedAccountIndex(uint32(childIdx)))
-	key, err := smbip32.Derive(HDPathToString(path), seed)
+	if kp.KeyType == typeLedger {
+		return pubkeyFromLedger(path, false)
+	} else if kp.KeyType == typeSoftware {
+		key, err := smbip32.Derive(HDPathToString(path), seed)
+		if err != nil {
+			return nil, err
+		}
+		return &EDKeyPair{
+			DisplayName: fmt.Sprintf("Child Key %d", childIdx),
+			Created:     common.NowTimeString(),
+			Private:     key[:],
```

Comment:
Here we are converting the array back to a slice - I think we can return the key as a slice in `Derive` 🙂